src/components/network-map-viewer/network/network-map.jsx

```diff
@@ -511,6 +516,120 @@ const NetworkMap = (props) => {
         setCentered(INITIAL_CENTERED);
     }, [mapLib?.key]);
 
+    const [features, setFeatures] = useState({});
+
+    useEffect(() => {
+        props.onFeaturesChanged(features);
+    }, [features]);
```

Comment:
What is the point of this?

Reply:
Agree. Why not calling `onFeaturesChanged` where you call `setFeatures`.
Effects procs between renders, then avoid using it for this I think.